### ISSUE TITLE
Implement Pyright

### ISSUE THREAD

**Issue by PhasecoreX** [SpartanJ/ecode#19]
Hello again!

I see in https://github.com/SpartanJ/eepp/blob/develop/bin/assets/plugins/lspclient.json that Python uses `pylsp` as the LSP client. I wanted to instead use Pyright, so I went ahead and modified my user `lspclient.json` like so:
```
{
  "config": {
    ...
  },
  "keybindings": {
    ...
  },
  "servers": [
    {
      "command": "pyright-langserver --stdio",
      "file_patterns": [
        "%.py$"
      ],
      "language": "python",
      "name": "pyright",
      "url": "https://github.com/microsoft/pyright"
    }
  ]
}
```
One or 2 versions ago, I'm pretty sure this worked correctly, but now I don't see any lines being highlighted, nor are autocompletes working. I see that `node /usr/bin/pyright-langserver --stdio` is started in htop when I open the editor as well.

Anything that I am missing? Command is wrong? Maybe some incompatibility between the LSP plugin and the linter plugin that displays errors? By the way, error lens looks very nice!

Actually I just noticed my linter settings, it looks like it already accounts for LSP diagnostics:
```
{
  "config": {
    "delay_time": "500ms",
    "disable_languages": [],
    "disable_lsp_languages": [],
    "enable_error_lens": true,
    "enable_lsp_diagnostics": true
  },
  "linters": []
}
```

**Comment by SpartanJ** [comment]
Hi @PhasecoreX! Something broke from `0.4.1` to `0.4.2` release. Some LSPs servers are complicated and break with very small changes. I'll take a look and keep you updated.

**Comment by SpartanJ** [comment]
@PhasecoreX I fixed locally, can you tell me the operating system you're using so I can send you a build with the fix?

**Comment by PhasecoreX** [comment]
Whoops, didn't see this soon enough. I'm using Arch Linux, the appimages specifically.

**Comment by SpartanJ** [comment]
Great, here's the patched AppImage version [ecode-linux-0.4.3-x86_64.zip](https://github.com/SpartanJ/ecode/files/10552550/ecode-linux-0.4.3-x86_64.zip).

**Comment by PhasecoreX** [comment]
Looks like it's working now, thank you!
